errors/src/main/java/com/palantir/conjure/java/api/errors/QosException.java

```diff
@@ -76,13 +95,21 @@ public static Throttle throttle(Duration duration) {
         return new Throttle(Optional.of(duration));
     }
 
+    public static Throttle throttle(Duration duration, QosReason reason) {
```

Comment:
For all of these as well, let's make QosReason the first argument in all factory methods where it's present